cmd/statusd/main.go

```diff
@@ -1,174 +1,134 @@
-	}
-	app.Flags = []cli.Flag{
-		ProdModeFlag,
-		NodeKeyFileFlag,
-		DataDirFlag,
-		NetworkIDFlag,
-		LogLevelFlag,
-		LogFileFlag,
+	if *logLevel != "" {
+		nodeConfig.LogLevel = *logLevel
 	}
-	app.Before = func(ctx *cli.Context) error {
-		runtime.GOMAXPROCS(runtime.NumCPU())
-		return nil
+	if *logFile != "" {
+		nodeConfig.LogFile = *logFile
 	}
-	app.After = func(ctx *cli.Context) error {
-		return nil
-	}
-}
 
-func main() {
-	if err := app.Run(os.Args); err != nil {
-		fmt.Fprintln(os.Stderr, err)
-		os.Exit(1)
-	}
-}
+	nodeConfig.LightEthConfig.Enabled = true
+	nodeConfig.RPCEnabled = *httpEnabled
+	nodeConfig.WhisperConfig.Enabled = *whisperEnabled
+	nodeConfig.SwarmConfig.Enabled = *swarmEnabled
 
-// makeApp creates an app with sane defaults.
-func makeApp(gitCommit string) *cli.App {
-	app := cli.NewApp()
-	app.Name = filepath.Base(os.Args[0])
-	app.Author = ""
-	//app.Authors = nil
-	app.Email = ""
-	app.Version = params.Version
-	if gitCommit != "" {
-		app.Version += "-" + gitCommit[:8]
+	// RPC configuration
+	if !*httpEnabled {
+		nodeConfig.HTTPHost = "" // HTTP RPC is disabled
 	}
-	app.Usage = "CLI for Status nodes management"
-	return app
+	nodeConfig.HTTPPort = *httpPort
+	nodeConfig.IPCEnabled = *ipcEnabled
+
+	return nodeConfig, nil
 }
 
-// makeNodeConfig parses incoming CLI options and returns node configuration object
-func makeNodeConfig(ctx *cli.Context) (*params.NodeConfig, error) {
-	nodeConfig, err := params.NewNodeConfig(
-		ctx.GlobalString(DataDirFlag.Name),
-		ctx.GlobalUint64(NetworkIDFlag.Name),
-		!ctx.GlobalBool(ProdModeFlag.Name))
-	if err != nil {
-		return nil, err
+// printVersion prints verbose output about version and config.
+func printVersion(config *params.NodeConfig, gitCommit, buildStamp string) {
+	version := params.Version
+	if gitCommit != "" && len(gitCommit) > 8 {
+		version += "-" + gitCommit[:8]
 	}
 
-	nodeConfig.NodeKeyFile = ctx.GlobalString(NodeKeyFileFlag.Name)
-
-	if logLevel := ctx.GlobalString(LogLevelFlag.Name); logLevel != "" {
-		nodeConfig.LogLevel = logLevel
+	fmt.Println(strings.Title(params.ClientIdentifier))
+	fmt.Println("Version:", params.Version)
+	if gitCommit != "" {
+		fmt.Println("Git Commit:", gitCommit)
 	}
-	if logFile := ctx.GlobalString(LogFileFlag.Name); logFile != "" {
-		nodeConfig.LogFile = logFile
+	if buildStamp != "" {
+		fmt.Println("Build Stamp:", buildStamp)
 	}
 
-	return nodeConfig, nil
-}
+	fmt.Println("Network Id:", config.NetworkID)
+	fmt.Println("Go Version:", runtime.Version())
+	fmt.Println("OS:", runtime.GOOS)
+	fmt.Printf("GOPATH=%s\n", os.Getenv("GOPATH"))
+	fmt.Printf("GOROOT=%s\n", runtime.GOROOT())
 
-// printNodeConfig prints node config
-func printNodeConfig(ctx *cli.Context) {
-	nodeConfig, err := makeNodeConfig(ctx)
-	if err != nil {
-		fmt.Printf("Loaded Config: failed (err: %v)", err)
-		return
-	}
-	nodeConfig.LightEthConfig.Genesis = "SKIP"
-	fmt.Println("Loaded Config: ", nodeConfig)
+	config.LightEthConfig.Genesis = "SKIP"
```

Comment:
Maybe change to `HIDDEN`? Also, I am not sure if this is needed when printing a version.

Reply:
@adambabik I have no idea, that code is from original implementation. This PR simply moves existing code into separate package, nothing else.